cpp/contracts/adapter.cpp

```diff
@@ -147,28 +152,104 @@ void adapter::freeuserdata(const name& account) {
    }
 }
 
+void adapter::settee(public_key pub_key, bytes attestation) {
+   print("set tee");
+   require_auth(get_self());
+   tee_pubkey _tee_pubkey(get_self(), get_self().value);
+
+   _tee_pubkey.get_or_create(
+      get_self(),
+      pam::tee{.key = public_key()}
+   );
+
+   _tee_pubkey.set(pam::tee{
+      .key = pub_key 
+   }, get_self());
+
+   // print("attestation: ")
+}
+
+void adapter::settopiczero(bytes chain_id, bytes topic_zero) {
+   require_auth(get_self());
+   check(topic_zero.size() == 32, "Expected 32 bytes emitter");
+   check(chain_id.size() == 32, "Expected 32 bytes chain_id");
+   mappings_table _mappings_table(get_self(), get_self().value);
+
+   auto mappings_itr = _mappings_table.find(pam::get_mappings_key(chain_id));
+
+   if (mappings_itr == _mappings_table.end()) {
+      _mappings_table.emplace(get_self(), [&](auto& row) {
+         row.chain_id = chain_id;
+         row.topic_zero = topic_zero;
+      });
+
+      print("Added a new mapping for chain_id: ", pam::get_mappings_key(chain_id));
+   } else {
+      _mappings_table.modify(mappings_itr, get_self(), [&](auto& row) {
+         row.topic_zero = topic_zero;
+      });
+
+      print("Updated the topic zero for chain_id: ", pam::get_mappings_key(chain_id));
+   }
+}
+
+void adapter::setemitter(bytes chain_id , bytes emitter) {
+   print("set emitter");
+   require_auth(get_self());
+   check(emitter.size() == 32, "Expected 32 bytes emitter");
+   check(chain_id.size() == 32, "Expected 32 bytes chain_id");
+   mappings_table _mappings_table(get_self(), get_self().value);
+
+   auto mappings_itr = _mappings_table.find(pam::get_mappings_key(chain_id));
+   if (mappings_itr == _mappings_table.end()) {
+      _mappings_table.emplace(get_self(), [&](auto& row) {
+         row.chain_id = chain_id;
+         row.emitter = emitter;
+      });
+
+      print("Added a new mapping for chain_id: ", pam::get_mappings_key(chain_id));
+   } else {
+      _mappings_table.modify(mappings_itr, get_self(), [&](auto& row) {
+         row.emitter = emitter;
+      });
+
+      print("Updated the emitter for chain_id: ", pam::get_mappings_key(chain_id));
+   }
+}
+
 void adapter::settle(const name& caller, const operation& operation, const metadata& metadata) {
    require_auth(caller);
 
    registry_adapter _registry(get_self(), get_self().value);
    auto idx_registry = _registry.get_index<adapter_registry_idx_token_bytes>();
    auto search_token_bytes = idx_registry.find(operation.token);
-
    check(search_token_bytes != idx_registry.end(), "invalid token");
 
-   checksum256 event_id;
-   check(pam::is_authorized(operation, metadata, event_id), "unauthorized");
```

Comment:
Why removing this? If it was for get_self() not being available within `is_authorized`, you can always pass it as arg like this
```cpp
pam::is_authorized(get_self(), operation, metadata, event_id);
```

Reply:
Then the tables should be accessible